### PR TITLE
Fix tensorflow batch job exit code on success

### PR DESCRIPTION
**Pull request by miguelvr**
checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

File: pkg/types/clusterconfig/cluster_config.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ type CoreConfig struct {
 	ImageOperator                   string `json:"image_operator" yaml:"image_operator"`
 	ImageManager                    string `json:"image_manager" yaml:"image_manager"`
 	ImageDownloader                 string `json:"image_downloader" yaml:"image_downloader"`
+	ImageKubexit                    string `json:"image_kubexit" yaml:"image_kubexit"`
```
Comment (deliahu) [inline pkg/types/clusterconfig/cluster_config.go:78]: There may be a few additional places where this should be added (search the code base for e.g. "istio-pilot", "istio_pilot", and "IstioPilot" to find them)

Comment (miguelvr) [reply pkg/types/clusterconfig/cluster_config.go:78]: I think those were removed

Comment (deliahu) [reply pkg/types/clusterconfig/cluster_config.go:78]: I think it's still missing in `create.md` (but that's the only remaining one I could find)

File: pkg/operator/operator/k8s.go [hunk 2]
```diff
@@ -971,6 +1006,28 @@ func getEnvVars(api *spec.API, container string) []kcore.EnvVar {
...
+		},
+		{
+			Name:  "KUBEXIT_GRAVEYARD",
+			Value: _kubexitGraveyardMountPath,
+		},
+	}
+
+	if deathDeps != nil {
+		envVars = append(envVars, kcore.EnvVar{
+			Name:  "KUBEXIT_DEATH_DEPS",
```
Comment (RobertLucian) [inline pkg/operator/operator/k8s.go:1023]: robert: what is this one for?

File: pkg/operator/operator/k8s.go [hunk 3]
```diff
@@ -369,10 +384,22 @@ func pythonPredictorContainers(api *spec.API, envVars []kcore.EnvVar) ([]kcore.C
...
+	return tensorFlowPredictorContainers(api, envVars, tfServingEnvVars, false)
 }
 
-func tensorFlowPredictorContainers(api *spec.API, envVars []kcore.EnvVar) ([]kcore.Container, []kcore.Volume) {
+func TensorFlowPredictorJobContainers(api *spec.API) ([]kcore.Container, []kcore.Volume) {
+	envVars := getEnvVars(api, APIContainerName)
+	envVars = append(envVars, getKubexitEnvVars(APIContainerName)...)
+
+	tfServingEnvVars := getEnvVars(api, _tfServingContainerName)
+	tfServingEnvVars = append(tfServingEnvVars, getKubexitEnvVars(_tfServingContainerName, APIContainerName)...)
```
Comment (RobertLucian) [inline pkg/operator/operator/k8s.go:397]: Should we also add `_neuronRTDContainerName` to `getKubexitEnvVars` as a param? If `compute.inf` > 0 and the tensorflow predictor is used, then the `_neuronRTDContainerName` will also be present as a sidecar.

File: images/tensorflow-serving-cpu/run.sh [hunk 4]
```diff
@@ -28,4 +28,9 @@ if [[ -n ${TF_MAX_BATCH_SIZE} && -n ${TF_BATCH_TIMEOUT_MICROS} ]]; then
 fi
 
 # launch TFS
-tensorflow_model_server "$@"
+
+if [ -f "/mnt/kubexit" ]; then
+  /mnt/kubexit tensorflow_model_server "$@"
```
Comment (RobertLucian) [inline images/tensorflow-serving-cpu/run.sh:33]: We should also do something like this to `images/neuron-rtd`.

File: pkg/operator/operator/k8s.go [hunk 5]
```diff
@@ -971,6 +1006,28 @@ func getEnvVars(api *spec.API, container string) []kcore.EnvVar {
...
 }
 
+func getKubexitEnvVars(containerName string, deathDeps ...string) []kcore.EnvVar {
+	envVars := []kcore.EnvVar{
+		{
+			Name:  "KUBEXIT_NAME",
+			Value: containerName,
+		},
+		{
+			Name:  "KUBEXIT_GRAVEYARD",
```
Comment (RobertLucian) [inline pkg/operator/operator/k8s.go:1016]: And these `KUBEXIT_*` vars are specific to what `kubexit` is expecting, right?

**Review by RobertLucian — COMMENTED** [review]
Generally LGTM except for an uncertainty about neuron-rtd.

File: pkg/operator/operator/k8s.go (hunk 2 above)
Comment (miguelvr) [reply pkg/operator/operator/k8s.go:1023]: for defining the death dependencies (e.g. tf serving has death dependency on the API container)

File: pkg/operator/operator/k8s.go (hunk 3 above)
Comment (miguelvr) [reply pkg/operator/operator/k8s.go:397]: makes sense

File: pkg/operator/operator/k8s.go (hunk 5 above)
Comment (miguelvr) [reply pkg/operator/operator/k8s.go:1016]: yes, they are

**Pull request closed without merging**